crates/db_schema/src/impls/captcha_answer.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+  schema::captcha_answer,
+  source::captcha_answer::CaptchaAnswer,
+  utils::{functions::lower, get_conn, naive_now, DbPool},
+};
+use diesel::{
+  delete,
+  dsl::exists,
+  insert_into,
+  result::Error,
+  select,
+  ExpressionMethods,
+  QueryDsl,
+};
+use diesel_async::RunQueryDsl;
+
+impl CaptchaAnswer {
+  pub async fn insert(pool: &DbPool, captcha: &CaptchaAnswer) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+
+    insert_into(captcha_answer::table)
+      .values(captcha)
+      .get_result::<Self>(conn)
+      .await
+  }
+
+  pub async fn check_captcha(pool: &DbPool, to_check: CaptchaAnswer) -> Result<bool, Error> {
+    let conn = &mut get_conn(pool).await?;
+
+    // delete any expired captchas
+    delete(captcha_answer::table.filter(captcha_answer::expires.lt(&naive_now())))
```

Comment:
Instead of doing this delete in the check, add it as a recurring job, in `scheduled_tasks.rs`, deleting every one with `published` older than 10 minutes older than now (and run every 10 minutes.)

Reply:
have done this, i've added a check to the bit that fetches the answer to ensure the answer it fetches isn't older than 10 mins still out of paranoia in case of scheduled tasks failing to run etc. i figure since it's not an expensive check the redundancy isn't too annoying?